packages/block-library/src/paragraph/style.scss

```diff
@@ -35,3 +35,8 @@ p.has-background {
 p.has-text-color a {
 	color: inherit;
 }
+
+// Prevent an empty P tag from collapsing, so it matches the backend.
+p:empty::before {
+	content: "\a0";
```

Comment:
This is `NO-BREAK SPACE`, the same as the `&nbsp;` pretty common in HTML. I'm not sure how screen readers handle this or whether there are considerations in the accessibility space to take into account.

There are some other characters that we could consider to ensure the `p` elements don't collapse, `ZERO WIDTH SPACE` `\200B` seems like a viable alternative. The advantage is that it's… zero width. It doesn't take up space.

https://user-images.githubusercontent.com/841763/104613953-0ab35d00-5688-11eb-9134-6ab01f00a06a.mov

Reply:
Naturally I got that wrong in the demo. The zero width space is below and the no-break space is above in the example. Zero width is the one that has… `0` width 🤦